ru/lessons/basics/composition.md

```diff
@@ -0,0 +1,236 @@
+---
+layout: page
+title: Композиция
+category: basics
+order: 8
+lang: ru
+---
+
+Мы все знаем из опыта, насколько неудобно хранить все функции в одном файле и одной области видимости. В этом уроке мы разберемся как группировать функции и определять специальный ассоциативный массив, называющийся `struct`, для более эффективной организации кода.
+
+## Содержание
+
+- [Модули](#section-1)
+  - [Атрибуты модулей](#section-2)
+- [Структуры](#section-3)
+- [Композиция](#section-4)
+  - [`alias`](#alias)
+  - [`import`](#import)
+  - [`require`](#require)
+  - [`use`](#use)
+
+## Модули
+
+Модули - лучший способ для организации функций по областям видимости. Кроме группировки функций они позволяют определять именованные и закрытые функции, которые мы рассмотрели в предыдущем уроке.
+
+Давайте рассмотрим простой пример:
+
+``` elixir
+defmodule Example do
+  def greeting(name) do
+    ~s(Hello #{name}.)
+  end
+end
+
+iex> Example.greeting "Sean"
+"Hello Sean."
+```
+
+Можно создавать вложенные модули, что позволяет еще сильнее разбивать функциональность на пространства имен:
+
+```elixir
+defmodule Example.Greetings do
+  def morning(name) do
+    "Good morning #{name}."
+  end
+
+  def evening(name) do
+    "Good night #{name}."
+  end
+end
+
+iex> Example.Greetings.morning "Sean"
+"Good morning Sean."
+```
+
+### Атрибуты модулей
+
+Атрибуты модулей зачастую используются в языке как константы. Давайте рассмотрим простой пример:
+
+```elixir
+defmodule Example do
+  @greeting "Hello"
+
+  def greeting(name) do
+    ~s(#{@greeting} #{name}.)
+  end
+end
+```
+
+Важно помнить, что есть несколько зарезервированных названий атрибутов в Elixir. Три самые часто используемые:
+
++ `moduledoc` — Документирует текущий модуль.
++ `doc` — Документация для функций и макросов.
++ `behaviour` — Использование OTP или определенного пользователем поведения.
+
+## Структуры
+
+Структура - специальный ассоциативный массив с определенным набором ключей и их значениями по умолчанию. Она должна быть определена в модуле, из которого и берет свое имя. Часто структура - это единственная вещь, определенная в модуле.
+
+Для определения структуры мы используем конструкцию `defstruct` вместе с списком с ключами полей и значений по умолчанию:
```

Comment:
Sorry, @brain-geek , I somehow didn't notice this commentary. 
I couldn't find anything simpler than using `и список ключей...` instead of `вместе со списком ключей...`.